src/main/java/games/strategy/engine/lobby/server/userDB/DBUserController.java

```diff
@@ -1,210 +1,112 @@
 package games.strategy.engine.lobby.server.userDB;
 
-import java.sql.Connection;
-import java.sql.PreparedStatement;
-import java.sql.ResultSet;
-import java.sql.SQLException;
-import java.sql.Timestamp;
-import java.util.logging.Level;
+import com.google.common.base.Preconditions;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
 import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
 
-import games.strategy.engine.framework.startup.ui.InGameLobbyWatcher;
-import games.strategy.util.MD5Crypt;
-import games.strategy.util.Util;
+/**
+ * Interface for interacting with database user operations. For example, add user, delete user..
+ */
+/*
+ * TODO: datasource migration note, from derby (java DB) to MySQL
+ *  This class is currently set up to write and read from multiple data sources.
+ *  This is a migration tool, to go from one data source to another. When we switch over completely
+ *  we can simplify and write/read to the primary datasource directly.
+ */
+public class DBUserController implements UserDao {
+  private static final Logger logger = Logger.getLogger(DBUserController.class.getName());
 
-public class DBUserController {
-  private static final Logger s_logger = Logger.getLogger(DBUserController.class.getName());
+  private static final Collection<UserDaoPrimarySecondary> implementations = Arrays.asList(
+      new DerbyUserController(),
+      new SqlUserController()
+  );
 
   /**
-   * @return if this user is valid.
+   * Stores a convenience reference to the authoritative DAO implementation.
+   * We use this datasource for reading, we always write to it, any errors during read/write are considered
+   * critical (on the other hand secondary DAO read/write errors can be logged and ignored)
    */
-  public String validate(final String userName, final String email, final String hashedPassword) {
-    if (email == null || !Util.isMailValid(email)) {
-      return "Invalid email address";
-    }
-    if (hashedPassword == null || hashedPassword.length() < 3 || !hashedPassword.startsWith(MD5Crypt.MAGIC)) {
-      return "Invalid password";
-    }
-    return validateUserName(userName);
-  }
+  private static final UserDao primaryDao;
 
-  public static String validateUserName(final String userName) {
-    // is this a valid user?
-    if (userName == null || !userName.matches("[0-9a-zA-Z_-]+") || userName.length() <= 2) {
-      return "Usernames must be at least 3 characters long and can only contain alpha numeric characters, -, and _";
-    }
-    if (userName.contains(InGameLobbyWatcher.LOBBY_WATCHER_NAME)) {
-      return InGameLobbyWatcher.LOBBY_WATCHER_NAME + " cannot be part of a name";
-    }
-    if (userName.toLowerCase().contains("admin")) {
-      return "Username can't contain the word admin";
-    }
-    return null;
+
+  static {
+    primaryDao = determinePrimaryDataSource();
   }
 
-  public static void main(final String[] args) throws SQLException {
```

Comment:
This main method was removed, seems to have been used for test at one point. Not having the dependency on DB here will be of help in further iterations.